docs/layout/dynamic-font-scaling.md

```diff
@@ -0,0 +1,205 @@
+# Dynamic Font Scaling
+
+Dynamic Font Scaling is a feature that allows users to choose the size of the text displayed on the screen. This helps users who need larger text for better readability, and it also accommodates users who can read smaller text.
+
+Dynamic Font Scaling is supported on Android, iOS, and iPadOS starting in Ionic v7.5.
+
+## Try It Out
+
+:::tip
+Be sure to try this on an Android, iOS, or iPadOS device.
+:::
+
+Follow the [Changing the Font Size on a Device](#changing-the-font-size-on-a-device) guide to set your preferred font size, and watch the text in the demo below grow or shrink according to your preferences.
+
+import DynamicFontScaling from '@site/static/usage/v7/layout/dynamic-font-scaling/index.md';
+
+<DynamicFontScaling />
+
+## Enabling Dynamic Font Scaling in Ionic
+
+:::info
+This feature is currently opt-in on iOS. However, it will be enabled by default starting in Ionic 8 at which point the following CSS will no longer be necessary.
+:::
+
+Dynamic Font Scaling is already enabled by default on Android. Developers can enable it on iOS by setting the following CSS globally:
+
+```css
+html {
+  --ion-dynamic-font: var(--ion-default-dynamic-font);
+}
+```
+
+Under the hood, Ionic sets the following CSS on iOS devices to enable Dynamic Font Scaling:
+
+```css
+html {
+  font: var(--ion-dynamic-font);
+}
+```
+
+Developers should ensure that the [typography.css](./global-stylesheets#typographycss) file is imported.
+
+## Using Dynamic Font Scaling
+
+### Integrating Custom Components
+
+Developers can configure their custom components to take advantage of Dynamic Font Scaling by converting any pixel font sizes to use [rem units](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#lengths). An easy way to convert from `px` to `rem` is to divide the pixel font size by the default browser font size, which is typically 16px.
+
+For example, if a component has a font size of `14px`, then this could be converted to `rem` by doing `14px / 16px = 0.875rem`.
+
+Developers should also audit their applications and make any other changes necessary to support larger font sizes. One thing to keep in mind is that the dimensions of your components may need to change to accommodate the larger font sizes. As a result, it may be necessary to change properties such as `width` and `height` to `min-width` and `min-height`. We also recommend having long text wrap to the next line instead of truncating to keep large text readable.
```

Comment:
Should we also mention `line-height` here? That seems to be one that gets set to `px` values sometimes.

Reply:
I mentioned `width` and `height` as an example -  I wasn't focused on calling out the most common properties. I can add it if you think it would help though.

Reply:
What if we linked to the documentation on [length](https://developer.mozilla.org/en-US/docs/Web/CSS/length) and just specified they should look over any CSS properties that accept length and change them from `px` to `rem` units?

Reply:
How does https://github.com/ionic-team/ionic-docs/pull/3111/commits/a9e12380a2df5fc6ec70a70f5d3ee0cefc99cae8 sound?

Reply:
This paragraph was originally less about updating px values to rem and more dealing with edge cases like the element needing to be allowed to grow in size. At the moment, the info is pretty much duplicated between this paragraph and the two above it.

Reply:
So my original goal with this section is to note that following:

1. Devs need to convert font size px units to rem
3. Devs may need to change other CSS values to account for the new font sizes

The change I made is meant to highlight point 3 whereas the first paragraph is meant to highlight point 1. Not every non-font-size CSS property that uses px will necessarily need to be converted to rem.

Reply:
I suppose I felt like specifically mentioning that `width` and `height` may need updating to `min-width` and `min-height` was more useful, since that's not a px/rem change and seems common. It's okay how it is now though.

Reply:
I like the addition of the link to the length docs. I think mentioning the `width`/`height` needing to change to `min-width`/`min-height` might be good for certain cases, like if they have a button with the following and want to support text wrapping:

```diff
button {
-  height: 48px;
+  min-height: 48px;
}
```

But in other cases they might have a rounded button like this that needs to scale proportionately:

```diff
button {
-  width: 48px;
-  height: 48px;
+  width: 3rem;
+  height: 3rem;
}
```

Reply:
OK I cleaned this up a bit. Thoughts on https://github.com/ionic-team/ionic-docs/pull/3111/commits/b0360dfb76ecd8bf971023f6b1d81d045fd61867?

Reply:
Much better 👍

Reply:
Thanks for the help both of you!